src/classes/RD2_ApiService.cls

```diff
@@ -253,4 +258,287 @@ public inherited sharing class RD2_ApiService {
+            RD2_QueueableService.enqueueOppEvalService(recurringDonationIds, recurringDonationIds);
+        }
+
+        return pauseData;
+    }
+
+    private List<npe03__Recurring_Donation__c> getRecurringDonationsById(Set<Id> rdIds) {
+        String soql = getRecurringDonationQuery();
+        return Database.query(soql);
+    }
+
+    private RecurringDonationSchedule__c createPauseSchedule(PauseObject pause, Id rdId) {
+        return new RD2_ScheduleService.PauseScheduleHandler().createPauseSchedule(
+            pause.statusReason,
+            pause.startDate,
+            pause.endDate,
+            rdId
+        );
+    }
+
+    private RecurringDonationSchedule__c deactivatePauseSchedule(RD2_RecurringDonation recurringDonation) {
+        RecurringDonationSchedule__c schedule = recurringDonation.getPauseSchedule();
+
+        if (schedule == null) {
+            return null;
+        }
+
+        new RD2_ScheduleService.PauseScheduleHandler().deactivatePauseSchedule(schedule);
+
+        return schedule;
+    }
+
+    private Set<Id> insertPauseSchedules(Map<Id, PauseObject> pauseData, List<RecurringDonationSchedule__c> schedules) {
+        List<Database.SaveResult> results = Database.insert(schedules, false);
+
+        return handleResultsAndReturnSuccessRD(results, pauseData, schedules);
+    }
+
+    private Set<Id> updatePauseSchedules(Map<Id, PauseObject> pauseData, List<RecurringDonationSchedule__c> schedules) {
+        List<Database.SaveResult> results = Database.update(schedules, false);
+
+        return handleResultsAndReturnSuccessRD(results, pauseData, schedules);
+    }
+
+    private Set<Id> handleResultsAndReturnSuccessRD(List<Database.SaveResult> results, Map<Id, PauseObject> pauseData, List<RecurringDonationSchedule__c> schedules) {
+        Set<Id> successfulProcessIds = new Set<Id>();
+
+        for (Integer i = 0; i < results.size(); i++) {
+            Id recurringDonationId = schedules[i].RecurringDonation__c;
+
+            if (results[i].isSuccess()) {
+                successfulProcessIds.add(recurringDonationId);
+            } else {
+                PauseObject pauseInfo = pauseData.get(recurringDonationId);
+                pauseInfo.addError(results[i].getErrors()[0].getMessage());
+            }
+        }
+
+        return successfulProcessIds;
+    }
+
+    public class PauseObject {
+
+        public Date startDate;
+        public Date endDate;
+        public String statusReason;
+        public Boolean unPause;
+        public Boolean isSuccess;
+        public String error;
+
+        public PauseObject addError(String errorString) {
+            this.isSuccess = false;
+            this.error = errorString;
+
+            return this;
+        }
+
+        public PauseObject populateDefaultValues() {
+            this.isSuccess = true;
+            this.unPause = (this.unPause == null)
+                ? false
+                : this.unPause;
+
+            return this;
+        }
```

Comment:
I'm not sure why this method is needed at all. You could simply set default values for the class vars as above.

Reply:
Since these objects get created by being deserialized, the default values get overridden with `null` if they're not present in the object that we're deserializing. We could alternatively treat null as false whenever we're using this variable to validate the pause request.

Reply:
Good point. We're not instantiating directly.